### ISSUE TITLE
Cluster restore fails with "127.0.0.1:6444: read: connection reset by peer"

### ISSUE THREAD

**Issue by ShylajaDevadiga** [k3s-io/k3s#4914]
**Environmental Info:**
K3s Version: 
k3s version v1.23.1+k3s1, v1.23.1-rc1+k3s2, 1.22.5-rc1+k3s2

Node(s) CPU architecture, OS, and Version: 
Any OS

Cluster Configuration:
3 servers, 1 agent 
3 servers

**Describe the bug:**
Restore from snapshot fails on a multinode setup

**Steps To Reproduce:**

- Create a 3 node cluster.
- Take snapshot
- Stop k3s on all nodes
- Restore from snapshot

**Expected behavior:**
Cluster should be restored from the snapshot

**Actual behavior:**
Restore fails with below error message
```
INFO[0006] etcd data store connection OK                
INFO[0006] ETCD server is now running                   
INFO[0006] k3s is up and running                        
WARN[0006] bootstrap key already exists                 
INFO[0006] Reconciling etcd snapshot data in k3s-etcd-snapshots ConfigMap 
WARN[0007] Deploy controller node name is empty or too long, and will not be tracked via server side apply field management 
INFO[0007] Failed to set etcd role label: failed to register CRDs: Get "https://127.0.0.1:6444/apis/apiextensions.k8s.io/v1/customresourcedefinitions": dial tcp 127.0.0.1:6444: connect: connection refused 
INFO[0007] Running load balancer 127.0.0.1:6444 -> [172.31.1.59:6443] 
ERRO[0007] failed to get CA certs: Get "https://127.0.0.1:6444/cacerts": read tcp 127.0.0.1:53328->127.0.0.1:6444: read: connection reset by peer 
ERRO[0009] failed to get CA certs: Get "https://127.0.0.1:6444/cacerts": read tcp 127.0.0.1:53336->127.0.0.1:6444: read: connection reset by peer 
ERRO[0011] failed to get CA certs: Get "https://127.0.0.1:6444/cacerts": read tcp 127.0.0.1:53346->127.0.0.1:6444: read: connection reset by peer 
WARN[0012] Deploy controller node name is empty or too long, and will not be tracked via server side apply field management 
```

**Additional context / logs:**
When the restore command is run one additional file is created on the server nodes, pointing the LB to the last etcd server that was added
```
/var/lib/rancher/k3s/agent/etc/k3s-agent-load-balancer.json
```

**Backporting**

- Needs backporting to 1.22.5

**Comment by brandond** [comment]
Adding some quick debug printing to the change from https://github.com/k3s-io/k3s/commit/05f1bc6e2a62a995564341b48e1191839f9b7e0f gives:
```console
INFO[0000] agentReady close check - SupervisorPort: 0 HTTPSPort: 6443
```

This happens because restore skips the normal initialization at https://github.com/k3s-io/k3s/blob/release-1.22/pkg/cli/server/server.go#L209-L211

**Comment by ShylajaDevadiga** [comment]
Fix is validated using rc2 on 1.22.5 as well as 1.23.1. Restore in current setup and fresh node setup is successful.
Scenarios covered
```
- Validate cluster-reset flag functionality

- Verify snapshot/restore functionality in a quorum loss scenario when snapshot is stored locally. 

- Create snapshot and store in s3 bucket

- Restore to any of the current nodes

- Restore to a new setup

```
